server.js

```diff
@@ -1,38 +1,161 @@
+  date_added: String,
+  release_year: Number,
+  rating: String, 
+  duration: String, 
+  listed_in: String,
+  description: String, 
+  type: String
+})
+
+const Movie = mongoose.model('Movie', movieSchema)
+
+// Function to start seeding our Database
+// will only run if RESET_DB environment variable is present and is true
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    // Starts by deleting any pre-existing Movie objects to prevent duplicates when running database
+    await Movie.deleteMany()
+
+    netflixData.forEach(async (show) => {
+      const newMovie = new Movie(show)
+      await newMovie.save()
+    })
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+app.get('/movies', async (req, res) => {
+  const { country, cast, show, page, per_page } = req.query
+
+  await Movie.updateMany(
+    { }, 
+    { 
+      $unset: { show_id: "" }
+    }
+  )
+  const movies = await Movie.aggregate([
+    { 
+      $match: {
+        country: {
+          $regex: new RegExp(country, "i")
+        }, 
+        cast: {
+          $regex: new RegExp(cast, "i")
+        },
+        type: {
+          $regex: new RegExp(show, "i")
+        }
+      }
+    }, 
+    {
+      $skip: Number((page + 1) * per_page + 1)
+    }/* , 
+    {
+      $limit: Number(per_page)
+    } */
+  ]) 
+
+  res.json({ length: movies.length, data: movies })
+})
+
+app.get('/movies/titles', async (req, res) => {
+  const title = await Movie.aggregate([
+    { $sort: { title: 1 } },
+    { $unset: ["_id", "director", "cast", "country", "date_added", "release_year", "rating", "duration", "listed_in", "description", "type", "__v"] }
+  ])
+  res.json({ length: title.length, data: title })
+})
+
+app.get('/movies/genres', async (req, res) => {
+  const genre = await Movie.find()
+  const filterGenres = genre
+    .map((item) => item.listed_in)
+    .map((item) => item.split(", "))
+  
+  let genresUnique = []
+  
+  filterGenres.forEach((outerItem) => {
+    // eslint-disable-next-line no-irregular-whitespace
+    outerItem.forEach((innerItem) => {
+      if (!genresUnique.includes(innerItem)) {
+        genresUnique.push(innerItem)
+      }  
+    })
+  })
+  res.json({ length: genresUnique.length, data: genresUnique })
+})
+
+app.get('/movies/years', async (req, res) => {
+  const year = await Movie.find()
+  const filterYear = year.map((item) => item.release_year)
+  let yearUnique = []
+
+  filterYear.forEach((item) => {
+    if (!yearUnique.includes(item)) {
+      yearUnique.push(item)
+    }
+    return yearUnique.sort().reverse()
+  })
+
+  res.json({ length: yearUnique.length, data: yearUnique })
+})
+
+app.get('/movies/years/latest', async (req, res) => {
```

Comment:
The data structure for this endpoint differs a bit from what I expected :O Since (I’m assuming) the parent endpoint in the hierarchy (/movies/years) is intended to include movies per year, I expected this endpoint to do the same, just that it fetches only recent years. Also, the movies aren’t currently sorted by release year, which too would be solved by "inheriting" the data structure of /movies/years :)